### PR TITLE
Large message database

### PR DESCRIPTION
**Pull request by mtaylor**

### CODE REVIEW

File: artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/file/JDBCSequentialFileFactory.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+   @Override
+   public void onIOError(Exception exception, String message, SequentialFile file) {
+
+   }
+
+   // Methods lifted from NIO Impl
+   @Override
+   public ByteBuffer allocateDirectBuffer(final int size) {
+      // Using direct buffer, as described on https://jira.jboss.org/browse/HORNETQ-467
```
Comment (clebertsuconic) [inline artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/file/JDBCSequentialFileFactory.java]: is this a copy & paste from the journal?

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/xa/BasicXaRecoveryTest.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ public void setUp() throws Exception {
       addressSettings.clear();
 
       if (storeType == StoreConfiguration.StoreType.DATABASE) {
-         configuration = createDefaultJDBCConfig().setJMXManagementEnabled(true);
```
Comment (clebertsuconic) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/xa/BasicXaRecoveryTest.java]: What this means? The JDBC wasn't really using JDBC on the test?

Comment (mtaylor) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/xa/BasicXaRecoveryTest.java]: Not sure what you mean.  I added boolean to createDefaultJDBCConfig to allow the test to pass in whether or not t configure the server to set up a netty connector (previously it always used Netty). It should be using JDBC for the test.

File: artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/file/JDBCSequentialFileFactory.java (hunk 1 above)
Comment (mtaylor) [inline artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/file/JDBCSequentialFileFactory.java]: Copy and Paste from the NIOSequentialFileFactory.  Actually this is a mistake I thought I had refactored this into a static method on the NIOSequentialFileFactory class and called it here.  I will update.  Thanks.

**Comment by mtaylor** [comment]
@clebertsuconic What happened to the last commit?

**Comment by clebertsuconic** [comment]
@mtaylor I squashed 152e572 into d19a7dd...

I preferred using an intermediate branch myself instead of asking you..  I was pretty sure about being the right thing to do.. let me explain:

Basically d19a7dd is breaking the testsuite, and 152e572 is fixing it. Well... you would have a commit with a broken code, and I would avoid that if possible... these committs belonged together so I squahsed them right before pushing upstream.

**Comment by mtaylor** [comment]
@clebertsuconic Ah OK.  The commits were separate, one adds large message support, the other enables XA tests for the JDBC journal.  Separate things.  But, as long as the code is in, I am happy.

**Pull request closed without merging**